script.js

```diff
@@ -1,56 +1,57 @@
-/* Copyright end */
-/* eslint-enable */
+const commonmark = require('commonmark')
 
 if (document.body.children.length === 1 &&
 document.getElementsByTagName('pre')[0] !== undefined) {
   fetch(location.href).then(response => {
     if (response.headers.get('Content-Type') === 'text/markdown') {
       response.text().then(text => {
+        let parser = new commonmark.Parser()
+        let renderer = new commonmark.HtmlRenderer()
+
+        let parsed = parser.parse(text)
+        let rendered = renderer.render(parsed)
+
+        let walker = parsed.walker()
+        while(walker.current.type !== 'heading') walker.next()
+        let title = (walker.current.type === 'heading') ? walker.current.firstChild.literal : 'Markdown Document' // Find relevant page content here?
```

Comment:
Does this work? Maybe the default title could be the file name?

Reply:
This works.

Default being file name sounds good.